client/pages/auth-form.jsx

```diff
@@ -75,20 +115,55 @@ export default class AuthForm extends React.Component {
     this.setState({ passwordType: nextType });
   }
 
+  clearError() {
+    this.setState({
+      usernameTooShort: false,
+      usernameTooLong: false,
+      usernameTaken: false,
+      passwordTooShort: false
+    });
+  }
+
   render() {
     const { handleChange, handleSubmit, togglePassword } = this;
-    const { username, password, passwordType, usernameTooShort, usernameTooLong, usernameTaken, passwordTooShort } = this.state;
+    const { username, password, passwordType, usernameTooShort, usernameTooLong, usernameTaken, passwordTooShort, invalidLogin } = this.state;
+    const { path } = this.context.route;
     const toggle = passwordType === 'password' ? 'images/eye-close.svg' : 'images/eye-open.svg';
+    let headingText, submitText, submitButtonClass, switchText, anchorText, anchorClass, anchorHref;
     let errorClass = 'auth-error-box';
     let errorMessage = '';
-    if (usernameTooShort || usernameTooLong || usernameTaken || passwordTooShort) {
-      errorClass += ' show';
-      if (usernameTooShort || usernameTooLong) {
-        errorMessage = 'Username must be 4-16 characters';
-      } else if (passwordTooShort) {
-        errorMessage = 'Password must be at least 6 characters';
-      } else if (usernameTaken) {
-        errorMessage = 'Username is already taken';
+    if (path === 'sign-up') {
+      headingText = 'SIGN UP';
+      submitText = 'Sign Up';
+      submitButtonClass = 'auth-submit-btn sign-up';
+      switchText = 'Already have an account?';
+      anchorText = 'LOG IN';
+      anchorClass = 'auth-switch-anchor sign-in';
+      anchorHref = '#sign-in';
+      if (usernameTooShort || usernameTooLong || usernameTaken || passwordTooShort) {
+        errorClass += ' show';
+        if (usernameTooShort || usernameTooLong) {
+          errorMessage = 'Username must be 4-16 characters';
+        } else if (passwordTooShort) {
+          errorMessage = 'Password must be at least 6 characters';
+        } else if (usernameTaken) {
+          errorMessage = 'Username is already taken';
+        }
+      }
+    } else if (path === 'sign-in') {
+      headingText = 'LOGIN';
+      submitText = 'Log In';
+      submitButtonClass = 'auth-submit-btn sign-in';
+      switchText = "Don't have an account?";
+      anchorText = 'SIGN UP';
+      anchorClass = 'auth-switch-anchor sign-up';
+      anchorHref = '#sign-up';
+      if (usernameTooShort || usernameTooLong || usernameTaken || passwordTooShort || invalidLogin) {
+        errorClass += ' show';
+        errorMessage = 'Invalid login';
+        if (usernameTooLong) {
+          errorMessage = 'Username must be 4-16 characters';
+        }
```

Comment:
This would be much cleaner if you separated these into `validateSignUp` and `validateSignIn` methods